node/subsystem-util/src/lib.rs

```diff
@@ -279,11 +285,13 @@ specialize_requests_ctx! {
 }
 
 /// From the given set of validators, find the first key we can sign with, if any.
-pub fn signing_key(validators: &[ValidatorId], keystore: &KeyStorePtr) -> Option<ValidatorPair> {
-	let keystore = keystore.read();
-	validators
-		.iter()
-		.find_map(|v| keystore.key_pair::<ValidatorPair>(&v).ok())
+pub async fn signing_key(validators: &[ValidatorId], keystore: CryptoStorePtr) -> Option<ValidatorId> {
```

Comment:
maybe it worth renaming to `signing_public_key` or `find_our_key`, previously it returned a key pair